Telecommunications Updated Machine UI

WondaMegapon opened the pull request:
<!-- Write **BELOW** The Headers and **ABOVE** The comments else it may not be viewable. -->
<!-- You can view Contributing.MD for a detailed description of the pull request process. -->

## About The Pull Request
Updates the interface for the telecommunication machines, allowing them to run using TGUI.

![UpdatedTCommUI](https://user-images.githubusercontent.com/18584493/87899615-5b1e6a80-ca17-11ea-9c9b-15c171c499c4.png)
<!-- Describe The Pull Request. Please be sure every change is documented or this can delay review and even discourage maintainers from merging your PR! -->

## Why It's Good For The Game
It visually enhances the experience of telecomms and turns a once daunting interface into something players can easily approach.

<!-- Please add a short description of why you think these changes would benefit the game. If you can't justify it in words, it might not be worth adding. -->

## Changelog
:cl:
add: Telecommunication machines now use a TGUI based interface
del: Removed HTML based telecommunication machine code
balance: Telecommunication machines can have their interfaces viewed without a multitool
server: Telecommunication machines now log changed settings
/:cl:

<!-- Both :cl:'s are required for the changelog to work! You can put your name to the right of the first :cl: if you want to overwrite your GitHub username as author ingame. -->
<!-- You can use multiple of the same prefix (they're only used for the icon ingame) and delete the unneeded ones. Despite some of the tags, changelogs should generally represent how a player might be affected by the changes rather than a summary of the PR's contents. -->

WondaMegapon committed:
Telecommunication Machine UI Overhaul

This replaces the UI of the tcomm machines from the old html system to the new TGUI system.

WondaMegapon committed:
Telecommunication Machine UI Overhaul

This replaces the UI of the tcomm machines from the old html system to the new TGUI system.

WondaMegapon committed:
Merge remote-tracking branch 'upstream/master' into tcomm-ui-update

WondaMegapon committed:
Updated TGUI.Bundle

WondaMegapon committed:
Fixing grammatical errors

stylemistake commented on tgui/packages/tgui/interfaces/Tcomm.js:
```diff
@@ -0,0 +1,245 @@
+      </Window.Content>
+    </Window>
+  );
+};
```
Missing final newline.

stylemistake commented on tgui/packages/tgui/interfaces/Tcomm.js:
```diff
@@ -0,0 +1,245 @@
+                        maxValue={maxfreq / 10}
+                        value={frequency}
+                        onChange={(e, value) => act(
+                          "tempfreq", { value: value })}
```
You can use a shorthand syntax for properties with the same name as variable:
```{ value }```

stylemistake commented on tgui/packages/tgui/interfaces/Tcomm.js:
```diff
@@ -0,0 +1,245 @@
+                    <Table.Cell>
+                      {RADIO_CHANNELS
+                        .find(channel => channel.freq === entry) && (
+                        <Box inline color={RADIO_CHANNELS
+                          .find(channel => channel.freq === entry)
+                          .color} ml={2}>
```
Each property should preferably be on its own line, to make it easier to read.

stylemistake reviewed (commented):
Looks pretty damn sweet for a first PR.

LemonInTheDark commented:
give this man good first pr for the love of god

stylemistake commented:
@LemonInTheDark "Good First PR" is traditionally given out to first contributors with "barely passable" contributions :D.

stylemistake commented on code/game/machinery/telecomms/machine_interactions.dm:
```diff
@@ -27,74 +29,176 @@
+	operator = user
+	ui = SStgui.try_update_ui(user, src, ui)
+	if(!ui)
+		ui = new(user, src, "Tcomm")
```
Change interface name to something proper, like `Telecomms`.

stylemistake commented on code/game/machinery/telecomms/machine_interactions.dm:
```diff
@@ -27,74 +29,176 @@
+		if("change_freq")
+			var/newfreq = params["value"]
+			if(newfreq)
+				if(findtext(num2text(newfreq), "."))
```
What happens if you pass an integer frequency with no dot after a division by 10? I'd say just stick to one format of frequency, and where necessary, divide or multiply by 10.

WondaMegapon committed:
Merge remote-tracking branch 'upstream/master' into tcomm-ui-update

WondaMegapon committed:
Housekeeping

Added final newline, shortened value-values, spread out boxes, renamed Tcomm to Telecomms, and made the backend use full numbers while the frontend used decimals.

Thanks for the feedback, y'all.

WondaMegapon commented:
Oops. Guess I'll have to figure out what went wrong in the morning.

stylemistake reviewed (dismissed):
Looks good to me. 👍

WondaMegapon committed:
Merge remote-tracking branch 'upstream/master' into tcomm-ui-update

WondaMegapon committed:
Merge remote-tracking branch 'upstream/master' into tcomm-ui-update

Okay, everything *should* be up to date now.
